widget/ui/src/components/Toast/Toast.Provider.tsx

```diff
@@ -0,0 +1,84 @@
+import type {
+  ProviderContext,
+  ProviderPropTypes,
+  ToastProps,
+  ToastType,
+} from './Toast.types';
+import type { PropsWithChildren } from 'react';
+
+import React, { createContext, useContext, useState } from 'react';
+import { createPortal } from 'react-dom';
+
+import { Toast } from './Toast';
+import { Wrapper } from './Toast.styles';
+
+const ToastContext = createContext<ProviderContext | undefined>(undefined);
+
+let id = 1;
```

Comment:
NIT: Consider id generator a separate function, so it can be easily replaced in future.

Toast.helpers.ts
```

function idGenerator(){
   let id = 1;
    return () => {
      id++;
       return id;
    }
}

```


`Toast.Provider.tsx` in component:
```
const generateId = useMemo(() => idGenerator(), []);
```

Whenerever you need:
```
generateId();
```

Reply:
fixed.